src/v/pandaproxy/schema_registry/test/compatibility_protobuf.cc

```diff
@@ -692,3 +692,15 @@ SEASTAR_THREAD_TEST_CASE(
       R"(syntax = "proto2"; message Simple { optional int32 id = 1; })",
       R"(syntax = "proto2"; message Simple { optional int32 id = 1; required int32 new_id = 2; })"));
 }
+
+SEASTAR_THREAD_TEST_CASE(
+  test_protobuf_compatibility_multiple_fields_moved_to_oneof) {
+    BOOST_REQUIRE(check_compatible(
+      pps::compatibility_level::forward,
+      R"(syntax = "proto3"; message Simple { int32 id = 1; })",
+      R"(syntax = "proto3"; message Simple { oneof wrapper { int32 id = 1; } })"));
+    BOOST_REQUIRE(!check_compatible(
+      pps::compatibility_level::forward,
+      R"(syntax = "proto3"; message Simple { int32 id = 1; int32 new_id = 2; })",
+      R"(syntax = "proto3"; message Simple { oneof wrapper { int32 id = 1; int32 new_id = 2; } })"));
```

Comment:
this is slightly surprising after reading my note in `protobuf.cc:L544-545` (I don't exactly remember what led me to that conclusion). Does the check also fail in the `backward` direction?

Reply:
oh, and if you can't construct a validating example, you may as well remove the comment 😉

Reply:
I removed this comment now, but I also didn't get what it meant. To me, that comment only makes sense in the other direction, ie. in this commit: https://github.com/redpanda-data/redpanda/pull/22798/commits/d2e710aad27ee6cef4d001b22d880198357ca803

> Does the check also fail in the backward direction?

No it doesn't. I've added a test for this now: https://github.com/redpanda-data/redpanda/pull/22798/commits/9c1670a8ed98b214a882fa79d94842fad9bb32b2

Reply:
>that comment only makes sense in the other direction, ie. in this commit: https://github.com/redpanda-data/redpanda/commit/d2e710aad27ee6cef4d001b22d880198357ca803

Yup, that's the one